zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Note.java

```diff
@@ -275,7 +275,8 @@ public Paragraph insertParagraph(int index) {
    */
   private void addLastReplNameIfEmptyText(Paragraph p) {
     String replName = lastReplName.get();
-    if (StringUtils.isEmpty(p.getText()) && StringUtils.isNotEmpty(replName)) {
+    if (StringUtils.isEmpty(p.getText()) && StringUtils.isNotEmpty(replName)
+        && factory.isBinding(getId(), replName)) {
```

Comment:
Would not it make more sense to move the method, that checks the existence of the note binding for current Note to the `Note.java` class?

Then the code could look like:

```java
static import org.apache.commons.lang.StringUtils;
//...
if (isEmpty(p.getText()) && isNotEmpty(replName) && this.isBinded(replName)) {
  //...
}

public boolean isBinding(String replName) {
  return factory.getInterpreter(this.id, replName) != null;
}
```